CHANGELOG.md

```diff
@@ -4,6 +4,41 @@
 
 ### Breaking changes
 
+You must make the following changes when you migrate to this release, or your service may break.
+
+#### Update package file paths
+
+To support package dependencies, we’ve moved our source files from `src` into `package/src`. In preparation for additional build targets, our package is now compiled into `package/dist`.
+
+##### Node.js and other bundlers
+
+Depending on the bundler you use, you may also need to make changes to your configuration file or assets paths to replace `node_modules/govuk-frontend` with `node_modules/govuk-frontend/dist`.
+
+This is typically where Node.js v12.19.0+ [`package.json` subpath patterns](https://nodejs.org/docs/latest-v18.x/api/packages.html#subpath-patterns) are not supported.
```

Comment:
I've put some detailed info in https://github.com/alphagov/govuk-frontend/pull/3498#discussion_r1179561715

Service teams using `require('govuk-frontend')` are least affected but newer bundler updates might map that to the ESM package export even though it's `require()` not `import`

Shall we ditch the pattern/wildcard mapping and keep it plain and simple?

i.e. Add the `dist/` prefix

Reply:
It's this sort of thing we can either support or enforce a switch into `dist/`

```js
const Button = require('govuk-frontend/govuk/components/button/button')
// const Button = require('govuk-frontend/dist/govuk/components/button/button')
```

Which would require a bundler that's aware of package exports

I've dug through some of the issues that popped up when `"exports"` was first added:

* https://github.com/alphagov/govuk-frontend/issues/2633
* https://github.com/alphagov/govuk-frontend/issues/2634
* https://github.com/alphagov/govuk-frontend/issues/2643
* https://github.com/alphagov/govuk-frontend/issues/2645

* https://github.com/alphagov/govuk-frontend/issues/711#issuecomment-1108805799

Reply:
I'd lean towards an explicit `dist` rather than hiding where things are through the `exports` field, but that needs wider discussion. I think that's something we can discuss as part of https://github.com/alphagov/govuk-frontend/issues/3482, depending on the different files we ship and how we give access to them we may come to different answers.

Reply:
Sounds good. That lowers our `"engines"` requirements too (Node.js v12.20.0 for wildcard package exports)

Reply:
Going to mark this as resolved. It got removed in https://github.com/alphagov/govuk-frontend/pull/3498/commits/ece24c1e3a4fa6434655fe77c37ccd7f5458642d + https://github.com/alphagov/govuk-frontend/pull/3498/commits/a36c7f5cb60e040755f2365d96c980cb15f8fa55